File: TaskFlow.Api/Models/Task.cs
namespace TaskFlow.Api.Models
{
    public class Task
    {
        public int Id { get; set; }
        public required string Title { get; set; }
        public string? Description { get; set; }
        public bool IsCompleted { get; set; }
        public DateTime CreatedAt { get; set; }
        public DateTime? DueDate { get; set; }

        public int UserId { get; set; }
        public virtual User User { get; set; } = null!;

        public int? CategoryId { get; set; }
        public virtual Category? Category { get; set; }
    }
}

File: taskflow-frontend/src/types/Task.ts
import type { TaskLocation } from './Location';

export interface Task {
    id: number;
    title: string;
    description?: string;
    isCompleted: boolean;
    createdAt: string;
    dueDate?: string | null;
    categoryId?: number | null;
    categoryName?: string;
    location?: TaskLocation | null;
}

export interface CreateTaskRequest {
    title: string;
    description?: string;
    dueDate?: string | null;
    categoryId?: number | null;
    categoryName?: string;
    location?: TaskLocation | null;
}

export interface UpdateTaskRequest {
    title?: string;
    description?: string;
    isCompleted?: boolean;
    dueDate?: string | null;
    categoryId?: number | null;
    categoryName?: string;
    location?: TaskLocation | null;
}